querier.go

```diff
@@ -354,11 +354,23 @@ func postingsForUnsetLabelMatcher(ix IndexReader, m labels.Matcher) (index.Posti
 		rit = append(rit, it)
 	}
 
+	merged := index.Merge(rit...)
+	// With many many postings, it's best to pre-calculate
```

Comment:
Nice, ping for me to update Thanos code as well: https://github.com/improbable-eng/thanos/blob/master/pkg/store/bucket.go#L1249